### PR TITLE
Categories refactor

### PR DESCRIPTION
**Pull request by craigh**
| Q                 | A
| ----------------- | ---
| Bug fix?          | no
| New feature?      | yes
| BC breaks?        | yes
| Deprecations?     | yes
| Fixed tickets     | -
| Refs tickets      | -
| License           | MIT
| Changelog updated | yes

## Description
refactor Categories module to use all FC methods. Update and strengthen tests of common use items like form elements.

## Todos
- [x] Tests
- [x] Documentation
- [x] Changelog

### CODE REVIEW

File: src/system/CategoriesModule/Form/Type/CategoriesType.php [hunk 1]
```diff
@@ -53,42 +54,14 @@ public function buildForm(FormBuilderInterface $builder, array $options)
 
         /** @var CategoryRegistryEntity[] $registries */
         foreach ($registries as $registry) {
```
Comment (craigh) [inline src/system/CategoriesModule/Form/Type/CategoriesType.php]: ping @Guite see below - much simpler 😉

Comment (Guite) [inline src/system/CategoriesModule/Form/Type/CategoriesType.php]: Awesome!

**Comment by craigh** [comment]
@Guite I think this is done. I know it is a lot of code but please review.

File: src/docs/Core-2.0/Categories/CategoriesType.md [hunk 2]
```diff
@@ -144,12 +144,16 @@ PageType
...
 ###Optional Options
 
  - `required` - (boolean) is the field required (default `true`).
  - `multiple` - (boolean) allow multiple selections (default `false`).
- 
\ No newline at end of file
+ - `includeGrandChildren` - @deprecated - use 'direct'.
+ - `direct` - (boolean) use only direct children or include all descendant generations (default `true`).
+ - `attr` - (array) attributes array for each select box (default `[]`).
+ - `csrf_protection` - (boolean) this overrides the default symfony behavior to default to `false`.
```
Comment (Guite) [inline src/docs/Core-2.0/Categories/CategoriesType.md]: For which purpose is the `csrf_protection` option?

File: src/style/core.css [hunk 3]
```diff
@@ -246,3 +246,17 @@ input:invalid, select:invalid, textarea:invalid, .form-error {
...
+.zikula-bootstrap-tab-container .tab-content {
+    padding: 1.5em 1em 0 0;
+    border: 1px solid #ddd;
+    border-top: 0;
+    margin-bottom: 1.5em;
+    background-color: #fff;
+}
+.zikula-bootstrap-tab-container .nav-tabs li {
+    margin: 0 0 -1px 0;
+}
```
Comment (Guite) [inline src/style/core.css]: Are these classes intended for common reusage? Should they be utilised by other modules, too?

File: src/system/CategoriesModule/Builder/EntitySelectionBuilder.php [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Zikula\CategoriesModule\Builder;
+
+use Zikula\Bundle\CoreBundle\HttpKernel\ZikulaHttpKernelInterface;
+
+class EntitySelectionBuilder
```
Comment (Guite) [inline src/system/CategoriesModule/Builder/EntitySelectionBuilder.php]: Is this something which should be used by modules with categorisable entities?

**Comment by Guite** [comment]
Generally: :100: 

Will existing usages of the old APIs completely break? Or will basic calls (getById, getByParentId, etc.) still work?

File: src/docs/Core-2.0/Categories/CategoriesType.md (hunk 2 above)
Comment (craigh) [inline src/docs/Core-2.0/Categories/CategoriesType.md]: I can't remember. I think I needed it for testing.

File: src/style/core.css (hunk 3 above)
Comment (craigh) [inline src/style/core.css]: yes

File: src/system/CategoriesModule/Builder/EntitySelectionBuilder.php (hunk 4 above)
Comment (craigh) [inline src/system/CategoriesModule/Builder/EntitySelectionBuilder.php]: no. it is used to build the `CategoryRegistryType` form

**Comment by craigh** [comment]
> Will existing usages of the old APIs completely break? Or will basic calls (getById, getByParentId, etc.) still work?

some will work. They should be all noted. All the UserCategories-related stuff is gone. other API stuff is indicated.

File: src/docs/Core-2.0/Categories/CategoriesType.md (hunk 2 above)
Comment (craigh) [inline src/docs/Core-2.0/Categories/CategoriesType.md]: I removed it...

**Pull request merged**